Src/Support/Google.Apis.Auth/GoogleJsonWebSignature.cs

```diff
@@ -97,38 +158,57 @@ public class GoogleJsonWebSignature
             }
 
-            // Verify signature
-            byte[] hash;
-            using (var hashAlg = SHA256.Create())
-            {
-                hash = hashAlg.ComputeHash(Encoding.ASCII.GetBytes($"{parts[0]}.{parts[1]}"));
-            }
-            bool verifiedOk = false;
-            foreach (var googleCert in await GetGoogleCertsAsync(clock, forceGoogleCertRefresh, certsJson))
+            // Validate the JWT, using process defined at:
+            // https://developers.google.com/identity/protocols/OpenIDConnect#validatinganidtoken
+
+            // Step 1: Verify that the ID token is properly signed by the issuer (Google, in our case).
+            if (!ignoreCertCheck)
             {
+                byte[] hash;
+                using (var hashAlg = SHA256.Create())
+                {
+                    hash = hashAlg.ComputeHash(Encoding.ASCII.GetBytes($"{parts[0]}.{parts[1]}"));
+                }
+                bool verifiedOk = false;
+                foreach (var googleCert in await GetGoogleCertsAsync(clock, settings.ForceGoogleCertRefresh, certsJson))
+                {
 #if NET45
                 verifiedOk = ((RSACryptoServiceProvider)googleCert).VerifyHash(hash, Sha256Oid, signature);
 #elif NETSTANDARD1_3
-                verifiedOk = googleCert.VerifyHash(hash, signature, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+                    verifiedOk = googleCert.VerifyHash(hash, signature, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
 #else
 #error Unsupported platform
 #endif
-                if (verifiedOk)
+                    if (verifiedOk)
+                    {
+                        break;
+                    }
+                }
+                if (!verifiedOk)
                 {
-                    break;
+                    throw new InvalidJwtException("JWT invalid, unable to verify signature.");
                 }
             }
-            if (!verifiedOk)
-            {
-                throw new InvalidJwtException("JWT invalid: unable to verify signature.");
-            }
 
-            // Verify iss, iat and exp claims
+            // Step 2: Verify that the value of iss in the ID token is equal to https://accounts.google.com or accounts.google.com.
             if (!ValidJwtIssuers.Contains(payload.Issuer))
             {
                 var validList = string.Join(", ", ValidJwtIssuers.Select(x => $"'{x}'"));
                 throw new InvalidJwtException($"JWT issuer incorrect. Must be one of: {validList}");
             }
+
+            // Step 3: Verify that the value of aud in the ID token is equal to your app’s client ID.
+            if (settings.Audience != null)
+            {
+                // Verify that all audiences in the JWT are trusted; i.e. in settings.Audience.
+                var trustedAudiences = new HashSet<string>(settings.Audience);
+                if (payload.AudienceAsList.Any(aud => !trustedAudiences.Contains(aud)))
```

Comment:
Perhaps:

```csharp
if (payload.AudienceAsList.Except(settingsAudience).Any())
```

(No need for creating the hash set manually. `Except` will do that.)